### PR TITLE
Add package sga.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/sga/package.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    def configure_args(self):
+        spec = self.spec	
+        args = [
+            '--with-sparsehash={0}'.format(spec['sparsehash'].prefix),
+            '--with-bamtools={0}'.format(spec['bamtools'].prefix),
+            '--with-jemalloc={0}'.format(spec['jemalloc'].prefix)
+        ]
+        return args
+
+    def install(self, spec, prefix):
```
Comment (davydden) [inline var/spack/repos/builtin/packages/sga/package.py]: you don't need install.

File: var/spack/repos/builtin/packages/sga/package.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        cd('src')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/sga/package.py]: you don't need install, setting `configure_directory` should be enough.

File: var/spack/repos/builtin/packages/sga/package.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool', type='build')
+
+    configure_directory = 'src'
+
+    def configure_args(self):
+        spec = self.spec	
+        args = [
```
Comment (davydden) [inline var/spack/repos/builtin/packages/sga/package.py]: i would write: 
```
return [
 ...
]
```

Comment (weijianwen) [inline var/spack/repos/builtin/packages/sga/package.py]: I'll fix them. I learned how to write specs from packages in spack. That's why it has a mix flavors.

**Pull request merged**